qt/scientific_interfaces/Muon/IALCDataLoadingModel.h

```diff
@@ -0,0 +1,47 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#pragma once
+
+#include "DllConfig.h"
+#include "IALCDataLoadingView.h"
+
+namespace MantidQt::CustomInterfaces {
+
+class MANTIDQT_MUONINTERFACE_DLL IALCDataLoadingModel {
+
+public:
+  virtual ~IALCDataLoadingModel() = default;
+  virtual void load(const IALCDataLoadingView *view) = 0;
+  virtual void cancelLoading() const = 0;
+  virtual Mantid::API::MatrixWorkspace_sptr exportWorkspace() = 0;
+  virtual bool checkCustomGrouping(const std::string &detGroupingType, const std::string &forwardGrouping,
+                                   const std::string &backwardGrouping) = 0;
+  virtual void updateAutoLoadCancelled() = 0;
+  virtual bool loadFilesFromWatchingDirectory(const std::string &firstFile, const std::vector<std::string> &files,
+                                              const std::string &runsText) = 0;
+  virtual std::string getPathFromFiles(std::vector<std::string> files) = 0;
```

Comment:
Can this parameter be `const &`, and the function be `const`?
```suggestion
  virtual std::string getPathFromFiles(const std::vector<std::string> &files) const = 0;
```